scripts/check_frontend_test_coverage.py

```diff
@@ -0,0 +1,213 @@
+    - File path
+    - File total lines
+    - File covered lines
+
+    Args:
+      file_path: string. The path of lcov file.
+
+    Returns:
+      An array with all tests filtered, including ony important data for the
+      diff.
+    """
+    with python_utils.open_file(file_path, 'r') as f:
+        tests_array = f.read().split('end_of_record')
+        tests_array_filtered = []
+
+        for test in tests_array:
+            lines = [line for line in test.splitlines() if filter_lines(line)]
+            if len(lines) > 0:
+                tests_array_filtered.append(lines)
+
+        return tests_array_filtered
+
+
+def build_tests_fully_coverage_dict():
+    """Build a dict with only fully covered files from develop branch.
+
+    Returns:
+        A dict containing file path, total lines and covered lines of each
+        tested file.
+    """
+    coverage_dict = {}
+    file_path = './tmp/dev-lcov.info'
+    file_exists = os.path.exists(file_path)
+
+    if file_exists is True:
+        tests = get_lcov_file_tests(file_path)
+        for lines in tests:
+            lf = lines[1].split(':')[1]
+            lh = lines[2].split(':')[1]
+
+            if lf == lh:
+                test_name = get_test_file_name(lines[0])
+                coverage_dict[test_name] = [
+                    int(lf),
+                    int(lh)
+                ]
+
+        return coverage_dict
+    else:
+        raise Exception('File at path {} doesn\'t exist'.format(file_path))
+
+
+def lcov_files_diff():
+    """Check if any 100% covered file had the coverage dropped."""
+    fully_covered_tests = build_tests_fully_coverage_dict()
+
+    file_path = './tmp/pr-lcov.info'
+    file_exists = os.path.exists(file_path)
+
+    if file_exists is True:
+        tests = get_lcov_file_tests(file_path)
+        for lines in tests:
+            test_name = get_test_file_name(lines[0])
+
+            if test_name in fully_covered_tests:
+                lf = lines[1].split(':')[1]
+                lh = lines[2].split(':')[1]
+
+                test = fully_covered_tests[test_name]
+                lf_test = test[0]
+
+                if lf_test == int(lf) and int(lf) != int(lh):
+                    sys.stderr.write(
+                        'The {} file is fully covered and it has '
+                        'decreased after the changes \n'.format(test_name))
+                    sys.exit(1)
+
+    else:
+        raise Exception('File at path {} doesn\'t exist'.format(file_path))
+
+
+def main():
+    """The first function to be executated."""
+    current_branch = subprocess.check_output([
+        'git',
+        'rev-parse',
+        '--abbrev-ref',
+        'HEAD'
+    ]).strip()
+
+    if current_branch != 'develop' and current_branch != 'master':
+        run_frontend_tests_script()
+
+        create_tmp_folder()
+
+        shutil.copyfile(
+            '../karma_coverage_reports/lcov.info',
+            './tmp/pr-lcov.info')
```

Comment:
For these file paths, use the `os.path` module to construct the paths, we also need to make sure this works on Windows platforms as well.

Reply:
Done